assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,145 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import { useSelect } from '@wordpress/data';
+import { ruleCard } from './styles';
+
+
+const Rule = ({
+  field,
+  loading,
+  newRule,
+  onAdd,
+  onDelete,
+  onUpdate,
+  reorderRule,
+  ruleIndex,
+  saving,
+  setOriginIndex,
+  setTargetIndex,
+  taxonomy,
+  term_id,
+  value,
+}) => {
+  const {
+    fields,
+    taxonomies,
+  } = wpLocalizedData;
+
+  const [rule, setRule] = useState({
+    field: Object.keys(fields)[0],
+    taxonomy: Object.keys(taxonomies)[0],
+    term_id: 0,
+    value: false,
+  });
+
+  // If existing rule, sync local state with incoming settings.
+  // Else set some default values.
+  useEffect(() => {
+    if(!newRule) {
+      setRule({
+        field: field,
+        taxonomy: taxonomy,
+        term_id: term_id,
+        value: value,
+      })
+    } else {
+      setRule({
+        field: Object.keys(fields)[0],
+        taxonomy: Object.keys(taxonomies)[0],
+        term_id: 0,
+        value: false,
+      })
+    }
+  },[taxonomy, term_id, field, value, newRule])
+
+  // const { loadingTerms, taxTerms } = useSelect((select) => ({
+  //   loadingTerms: select('core/data').isResolving('core', 'getEntityRecords', ['taxonomy', 'category']),
+  //   taxTerms: select('core').getEntityRecords('taxonomy', 'category') || [],
+  // }));
+  // if(!loadingTerms) {
+  //   console.log(taxTerms)
+  // }
+
+  return (
+    <div
+      className="rule-wrapper"
+      draggable={!newRule}
```

Comment:
Don't worry about this. A new rule appended to the end should be able to be dragged even if not all fields are set yet.